aten/src/ATen/function_wrapper.py

```diff
@@ -732,6 +742,23 @@ def process_option(option, output_options):
             top_env['type_method_definitions'].append(
                 TYPE_METHOD_DEFINITION_ABSTRACT.substitute(env))
 
+            # Set up type promotion, if used
+            unify_types = ''
+            has_side_effects = any('Tensor' in f and 'const' not in f
+                                   for f in option['formals'])
+            if option.get('zero_dim_dispatch_when_scalar') and not has_side_effects:
```

Comment:
Maybe comment on why we only do this when zero_dim_dispatch_when_scalar?  It doesn't seem like there's any particular reason we have to restrict this behavior to that case, but we are just re-establishing the old behavior, right?

Reply:
Yes, that's just to fix the cases that previously were working. We can enable that for a wider class of ops later if we want